### PR TITLE
Ability to sort search results by sortable fields (sortby). - IN PROGRESS

### PR DESCRIPTION
**Pull request by deivarezserve**
I can't quite figure out why this is not working, can anybody provide some insight?

### CODE REVIEW

File: src/Query/Builder.php [hunk 1]
```diff
@@ -22,6 +22,7 @@ class Builder implements BuilderInterface
     protected $inKeys = '';
     protected $numericFilters = [];
     protected $geoFilters = [];
+    protected $sortBy = [];
```
Comment (ethanhann) [inline src/Query/Builder.php]: If I am reading the doc for [SORTBY](http://redisearch.io/Sorting/#specifying-sortby) correctly, you can only have one SORTBY statement per query. That means that $sortBy should be a string, not an array.

File: src/Query/Builder.php [hunk 2]
```diff
@@ -122,7 +129,8 @@ public function search(string $query, bool $documentsAsArray = false): SearchRes
                 $this->numericFilters,
                 explode(' ', array_reduce($this->geoFilters, function ($previous, $next) {
                     return $previous . $next;
-                }))
+                })),
+                $this->sortBy
```
Comment (ethanhann) [inline src/Query/Builder.php]: After you make $this->sortBy a string, you probably need to explode the string into a list of arguments here: explode(' ', $this->sortBy)

File: src/Query/Builder.php (hunk 1 above)
Comment (deivarezserve) [reply src/Query/Builder.php]: Fixed

File: src/Query/Builder.php (hunk 2 above)
Comment (deivarezserve) [reply src/Query/Builder.php]: Fixed

**Comment by ethanhann** [comment]
Btw, I created this issue to track API coverage: https://github.com/ethanhann/redisearch-php/issues/15

**Comment by deivarezserve** [comment]
Hi, Ethan, thank you. When you say FT.SEARCH implementation is Incomplete, can you please elaborate?
 What are the features missing? The reason I'm asking is that I'm trying to use a quoted query and it is failing, and I don't really understand why...

The redis-cli query is this one:
127.0.0.1:6379> ft.search menu-api_dev:restaurant-index " pizza @city:Nashville @state:3967 @country:225" filter status 1 1 limit 0 25 nocontent
1) (integer) 1
2) "12519833"

And when I do the same using the library, I get this:

Args:
array (
  0 => 'FT.SEARCH',
  1 => 'menu-api_dev:restaurant-index',
  2 => '" Pizza @city:Nashville @state:3967 @country:225"',
  3 => 'LIMIT',
  4 => '0',
  5 => '25',
  6 => 'FILTER status 1 1',
)

Response:
Syntax error at offset 8 near 'city

**Comment by deivarezserve** [comment]
Conflicts fixed.

**Comment by ethanhann** [comment]
I meant that not all the [FT.SEARCH parameters](http://redisearch.io/Commands/#parameters_4) are available in the library yet.

There might be issues with complex queries. I don't have any tests for them yet. The search query string might have to be tokenized and passed as individual array arguments to the raw command method. I'll write some tests and take a look.

**Comment by ethanhann** [comment]
The rawCommand function doesn't like that there is a ":" in the quoted query. I'm not sure why yet.

**Comment by ethanhann** [comment]
It appears that you are mixing quoted queries and specific field selection syntax which is not considered valid by the RediSearch module.

I'll describe what I am seeing for the following index.

```
FT.CREATE RestaurantIndex SCHEMA city TEXT WEIGHT 1 state TEXT WEIGHT 1 country TEXT WEIGHT 1 name TEXT WEIGHT 1
FT.ADD RestaurantIndex bcda 1 FIELDS name pizza city Nashville state 3967 country 255
FT.ADD RestaurantIndex bcdb 1 FIELDS name hamburger city Miami state 2000 country 300
```

Everything after pizza is ignored by the query parser when the search query is not quoted and only "pizza" is searched for. Notice that the city is "Miami" in the query, but "Nashville" in the results.

```
Query:
FT.SEARCH RestaurantIndex pizza @city:Miami @State:3967 @Country:255
Results:
1) (integer) 1
2) "bcda"
3) 1) "name"
   2) "pizza"
   3) "city"
   4) "Nashville"
   5) "state"
   6) "3967"
   7) "country"
   8) "255"
```

This query is not actually a "quoted" (i.e. exact match) query even though it has quotes.

```
Query:
FT.SEARCH RestaurantIndex "pizza @city:Miami @State:3967 @Country:255"
Results:
1) (integer) 0
```

If we change the city to "Nashiville" we get a result, because all the parts of the query are considered.

```
Query:
FT.SEARCH RestaurantIndex "pizza @city:Nashville @State:3967 @Country:255"
Results:
1) (integer) 1
2) "bcda"
3) 1) "name"
   2) "pizza"
   3) "city"
   4) "Nashville"
   5) "state"
   6) "3967"
   7) "country"
   8) "255"
```

This is actually a quoted query. Notice that the error is the same as what you are seeing when you try to do a quoted query in the module.

```
Query:
FT.SEARCH RestaurantIndex '"pizza @city:Nashville @State:3967 @Country:255"'
Results:
(error) Syntax error at offset 7 near 'city'
```

Perhaps you are looking for something like this?

```
RestaurantIndex '"pizza party" @city:Nashville @State:3967 @Country:255'
```

**Comment by ethanhann** [comment]
Related: https://github.com/RedisLabsModules/RediSearch/issues/154

**Comment by deivarezserve** [comment]
Thank you, @ethanhann , for your explanation. I think I don't really need the quotes then.

**Comment by deivarezserve** [comment]
Can you merge this PR or is there something missing?

**Pull request merged**